src/v/storage/segment.cc

```diff
@@ -246,7 +246,6 @@ ss::future<> segment::flush() {
     });
 }
 ss::future<> segment::do_flush() {
-    _generation_id++;
```

Comment:
I think it is modifying the segment, because it's flushing all the buffered data in segment_appender into the underlying segment

Reply:
agree, i will drop that change